### DISCUSSION TITLE
package.json#main examples inconsistent

### DISCUSSION THREAD

**Discussion by o-az** [vercel/turborepo#2983] [Help] [answered]
In the docs website, this example says to add `main` and `types` as `src/index.ts`: https://turbo.build/repo/docs/handbook/sharing-code/internal-packages#4-fix-main-and-types

In this example in the examples directory, `main` is `./dist/index.js`

https://github.com/vercel/turbo/blob/22773fc9c5f03adbd543d6de4f88b4ca0120056a/examples/with-docker/packages/logger/package.json#L5-L7

Is there a reason why they're different and does turbo recommend one over the other? The implications are mostly from the DX side.

**Comment by mehulkar** [comment, marked as answer]
There's no strong reason. `./dist/index.js` is generally used when a package has a compile step, so the src is not what's published. The examples show a variety of ways to set things up, but one is not more right than another. If you run into an issue running one of the examples, please open an Issue!

**Comment by o-az** [comment]
@mehulkar what is the meaning if the `source` field? What does it do?

**Reply by mehulkar** [reply]
I'm not sure where that comes from. The JS ecosystem has a lot of conventions around these fields and it could be some tool that we have in the example (or used to have at some point) was using it. Once you use the example, you could definitely try removing it.
